### PR TITLE
BlueLiv Crimeserver Parser tor type enhancements

### PR DESCRIPTION
**Pull request by navtej**
Fixes :
1/ For TOR_IP type it sets the classification to proxy
2/ Fixes the exception which is raised when tor ip is attempted to added as url

### CODE REVIEW

File: intelmq/bots/parsers/blueliv/parser_crimeserver.py [hunk 1]
```diff
@@ -25,14 +27,25 @@ def process(self):
...
+                    tor_node = True
             if 'url' in item:
-                event.add('source.url', item['url'])
+                # crimeserver reports tor ips in url as well, skip those
+
+                if not tor_node:
+                    event.add('source.url', item['url'])
+                else:
+                    valid_ip = IPAddress.is_valid(item['url'])
+                    if not valid_ip:
```
Comment (wagner-certat) [inline intelmq/bots/parsers/blueliv/parser_crimeserver.py]: What should happen if it's a valid IP addess?

File: intelmq/bots/parsers/blueliv/parser_crimeserver.py [hunk 2]
```diff
@@ -25,14 +27,25 @@ def process(self):
...
+                    event.add('source.tor_node', True)
+                    tor_node = True
             if 'url' in item:
-                event.add('source.url', item['url'])
+                # crimeserver reports tor ips in url as well, skip those
+
+                if not tor_node:
+                    event.add('source.url', item['url'])
+                else:
+                    valid_ip = IPAddress.is_valid(item['url'])
```
Comment (wagner-certat) [inline intelmq/bots/parsers/blueliv/parser_crimeserver.py]: Either use `event.is_valid(...)` or `event.add(..., raise_failure=False)` which makes the code shorter (and remove the harmonization import).

File: intelmq/bots/parsers/blueliv/parser_crimeserver.py (hunk 1 above)
Comment (navtej) [inline intelmq/bots/parsers/blueliv/parser_crimeserver.py]: code falls to the ip check block and ip is added

```
            if 'ip' in item:
                event.add('source.ip', item['ip'])
```

For TOR cases the feed provides IP in both IP and URL fields.

File: intelmq/bots/parsers/blueliv/parser_crimeserver.py (hunk 2 above)
Comment (navtej) [inline intelmq/bots/parsers/blueliv/parser_crimeserver.py]: changed the logic by using `raise_failure`, the flg is set for tor_node and `raise_failure` set t False for tor_nodes. Much simpler code.

**Comment by navtej** [comment]
@wagner-certat Abt the test case, can you guide me here what do I add? the TOR_IP entry in .data file?

**Comment by wagner-certat** [comment]
I pushed reorganized and extended tests and a bugfix. Please check if the data in the tests is realistic.

**Comment by wagner-certat** [comment]
Fixed the merge conflicts after merging #922 too

**Comment by wagner-certat** [comment]
No feedback - merged

**Pull request merged**